### ISSUE TITLE
[Error [TurbopackInternalError]: Next.js package not found

### ISSUE THREAD

**Issue by Nizodevlo** [vercel/next.js#74226]
### Link to the code that reproduces this issue

https://github.com/Nizodevlo/Redux.git

### To Reproduce

1. first initialized the project with: npx create-next-app@latest ./ --use-pnpm
2. then create this folder structure: 
![{B3B5364A-27C5-4A46-A42E-58E0004F95A0}](https://github.com/user-attachments/assets/80bf259a-e1ec-4a3e-b1de-c709c1a8ce9c)
3. Add the Store provider to the root layout:
![{3DE31F11-E787-4E73-9D8D-7D9E532B3537}](https://github.com/user-attachments/assets/c2b60810-a638-4e1a-bd86-be34f3e71f66)
4.execute: pnpm dev
5. here's waht happend:
![{AB074F7A-97A3-4B66-948A-19D9917C3381}](https://github.com/user-attachments/assets/e5f80c5d-a779-4f97-a539-2a7eeb1ab03c)

here's the report log:
![{8AFA4125-7A33-4BF1-9C05-5AB127F32166}](https://github.com/user-attachments/assets/cbf2ac3b-53fe-4df0-bb4e-46d68abb6cfa)

### Current vs. Expected behavior

I tryed to use the store provider in root layout:
![{D508A0E8-8B58-4725-BE99-02D32E9F2670}](https://github.com/user-attachments/assets/ec7479a4-c732-4f49-badf-ab9d065968ee)

### Provide environment information

```bash
Operating System:
  Platform: win32
  Arch: x64
  Version: Windows 11 Home Single Language
  Available memory (MB): 16076
  Available CPU cores: 8
Binaries:
  Node: 22.12.0
  npm: 11.0.0
  Yarn: N/A
  pnpm: 9.15.1
Relevant Packages:
  next: 15.1.1-canary.16 // Latest available version is detected (15.1.1-canary.16).
  eslint-config-next: 15.1.2
  react: 19.0.0
  react-dom: 19.0.0
  typescript: 5.7.2
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

Turbopack

### Which stage(s) are affected? (Select all that apply)

next dev (local)

### Additional context

I've used the canary version 15.1.1-canary.16 yet still the same issue

**Comment by Ivor808** [comment]
I have also been running into this issue on fresh next builds even without redux.

**Comment by Ivor808** [comment]
@Nizodevlo what error do you get if you build without the --trubopack flag in the pnpm dev cmd?

**Comment by Ivor808** [comment]
I may have figured this out. I think the issue is the project is started with "npx" which also runs the install command. If you run pnpm install on top of this, there are node_module collisions.

**Comment by Nizodevlo** [comment]
@Ivor808 after I removed the flag, it showed the ui for a second but it disappeared.
Here's the error : 
![{3326BEDA-B20C-4E8F-9996-BD04CEBEF909}](https://github.com/user-attachments/assets/d6c1e096-542a-4c4a-83e4-b9ceffa1ead4)
It said that I'm using outdated canary version but I'v used this cmd in docs: npm i next@canary 
Now I will try it again if its not working. I'm gonna use this cmd: npm i next@latest

After I use "npm i next@latest" it showed different error:
![{025895AD-01C3-4D37-99E2-3551B9DB52F6}](https://github.com/user-attachments/assets/61900900-8d7d-4c0f-bd26-1ffee26b748b)
here's the terminal:
![{296A4082-BF12-4322-8115-43518C3ABFDC}](https://github.com/user-attachments/assets/b7f98056-e439-4b31-bb0c-42c0c1f511af)


I will try again fresh using npm not pnpm to see if there is a difference

**Comment by Nizodevlo** [comment]
Update:

-I've used just redo everything here's the steps:
1.delete node_modules folder: rm -rf node_modules package-lock.json
2.clear the cache: npm clear cache --force
3.reinstall the dependencies with: npm install

Now it workings:
![{CE0274F9-893D-45BE-B01F-2740A601A380}](https://github.com/user-attachments/assets/bfe62b49-f737-4075-9c4a-0229c9f9787e)
![{5B4B2BD2-FB01-4CDC-9077-ADCB9DCC65D3}](https://github.com/user-attachments/assets/8f98c253-44d9-474e-9f44-f37796f5fc01)


I think the problem is coming from pnpm. Just my opinion cuz I'm 1year to development

**Comment by Nizodevlo** [comment]
@Ivor808 try using just npm and see what happend cuz it fixed my problem

**Comment by bburgess19** [comment]
@Nizodevlo this worked for me! Thanks.

Though, make sure to use

```npm *cache* clear --force```

**Comment by Ivor808** [comment]
yep using just npm fixed my issue. Anyone know if this issue is tracked with pnpm or next?

**Comment by Nizodevlo** [comment]
No problem brothers.
@bburgess19 thx for the info I always mess up in commands

**Comment by romka-best** [comment]
I fixed it via line "outputFileTracingRoot: path.join(__dirname, '../../')," in next.config.ts file. I'm using pnpm 10.1.0 and next: 15.1.6. Maybe I can help who couldn't fix with @Nizodevlo solution like me

```
import type { NextConfig } from 'next';

const nextConfig: NextConfig = {
  outputFileTracingRoot: path.join(__dirname, '../../'),
};

export default nextConfig;
```

**Comment by lockhaicttv** [comment]
For everyone got this issue, remove pack-log.json is the last way save me

**Comment by timneutkens** [comment]
The provided reproduction is private so I can't check it. But the most common reason for this error is that the project root is wrong / can't be detected. 

I've added docs for why and how to fix here: https://nextjs.org/docs/app/api-reference/config/next-config-js/turbopack#root-directory

We're going to improve the error itself as well but that's tracked separately. Going to close this issue.

**Comment by monarchwadia** [comment]
I just removed Turbopack 

Before:
```
  // package.json
  "scripts": {
    "dev": "next dev --turbopack",
```

After:
```
  // package.json
  "scripts": {
    "dev": "next dev",
```
